runelite-client/src/main/java/net/runelite/client/util/StackFormatter.java

```diff
@@ -133,54 +139,23 @@ else if (quantity < 10_000_000)
 	 * appear in RuneScape. (with decimals)
 	 * <p>
 	 * This differs from quantityToRSStack in that it displays
-	 * decimals. Ex: 27100 is 27,1k (not 27k)
+	 * decimals. Ex: 27100 is 27.1k (not 27k)
 	 * <p>
-	 * This uses the NumberFormat singleton instead of the
-	 * NUMBER_FORMATTER variable to ensure the UK locale.
 	 *
 	 * @param quantity The quantity to convert.
 	 * @return The stack size as it would appear in RS, with decimals,
 	 * with K after 100,000 and M after 10,000,000
 	 */
 	public static String quantityToRSDecimalStack(int quantity)
 	{
-
-		if (quantity < 10_000)
-		{
-			return Integer.toString(quantity);
-		}
-		else if (quantity < 1_000_000)
-		{
-			if (quantity % 1000 == 0)
-			{
-				return quantity / 1000 + "K";
-			}
-			return NUMBER_FORMATTER.format(quantity).substring(0, Integer.toString(quantity).length() - 1) + "K";
-		}
-		else if (quantity < 10_000_000)
+		String quantityStr = String.valueOf(quantity);
+		if (quantityStr.length() <= 4)
 		{
-			if (quantity % 1_000_000 == 0)
-			{
-				return quantity / 1_000_000 + "M";
-			}
-			return NUMBER_FORMATTER.format(quantity).substring(0, Integer.toString(quantity).length() - 4) + "M";
-		}
-		else if (quantity < 1_000_000_000)
-		{
-			if (quantity % 1_000_000 == 0)
-			{
-				return quantity / 1_000_000 + "M";
-			}
-			return NUMBER_FORMATTER.format(quantity).substring(0, Integer.toString(quantity).length() - 4) + "M";
-		}
-		else
-		{
-			if (quantity % 1_000_000_000 == 0)
-			{
-				return quantity / 1_000_000_000 + "B";
-			}
-			return NUMBER_FORMATTER.format(quantity).substring(0, Integer.toString(quantity).length() - 7) + "B";
+			return quantityStr;
 		}
+
+		int power = (int) StrictMath.log10(quantity);
```

Comment:
Is it necessary to use `StrictMath` here?

Reply:
That being said I'll update the PR with `Math` when I get home tonight, as there is no scenario in which we need `StrictMath`s strict floating point accuracy